hs_core/templatetags/hydroshare_tags.py

```diff
@@ -238,3 +248,67 @@ def normalize_human_name(name):
 def display_name_to_class(value):
     """ Converts an aggregation display name to a string that is usable as a CSS class name """
     return value.replace(" ", "_").lower()
+
+
+@register.filter
+def creator_json_ld_element(crs):
+    """ return json ld element for one creator for schema.org script embedded on resource landing page"""
+    crs_array = []
+    for cr in crs:
+        cr_dict = {}
+        urls = []
+        if cr.email:
+            cr_dict["email"] = cr.email
+        if cr.address:
+            cr_dict["address"] = {
+                "@type": "PostalAddress",
+                "streetAddress": cr.address
+            }
+        if cr.name:
+            cr_dict["@type"] = "Person"
+            cr_dict["name"] = name_without_commas(cr.name)
+            if cr.organization:
+                affl_dict = {
+                    "@type": "Organization",
+                    "name": cr.organization
+                }
+                cr_dict["affiliation"] = affl_dict
+        else:
+            cr_dict["@type"] = "Organization"
+            cr_dict["name"] = cr.organization
+
+        if cr.description:
+            if cr.name:
+                # append www.hydroshare.org since schema.org script is only embedded in production
+                urls.append("https://www.hydroshare.org" + cr.description)
+            else:
+                # organization
+                urls.append(cr.description)
+        if cr.homepage:
+            urls.append(cr.homepage)
+        if cr.identifiers:
+            for k in cr.identifiers:
+                urls.append(cr.identifiers[k])
+        if len(urls) == 1:
```

Comment:
Why are you not including the list when only one url exists?  I'd think it's important to be consistent and always use a list.  This seems intentional, maybe there is something I'm not aware of.

Reply:
Yes, it is intentional. Otherwise, the returned json-ld element will have extra list delineators which don't conform to schema.org syntax requirements.

Reply:
It seems odd to remove the list if there is only one element, but I'll take your word on it.  Any idea why that is a syntax requirement?

Reply:
I am not sure why that is a syntax requirement, but from the examples created by @horsburgh, it appears to be the case. For example, for the identifier element, when there is only one identifier element, it is a dict as shown here: https://gist.github.com/horsburgh/29c44a4b526603bc174605189d0001e3#file-suggested_hydroshare_schema-org_encoding-json-L22-L33, however, when there are multiple identifier elements, it became a list as shown here: https://gist.github.com/horsburgh/29c44a4b526603bc174605189d0001e3#file-suggested_hydroshare_schema-org_encoding-json-L129-L151.